app/elm/Combat.elm

```diff
@@ -3,20 +3,21 @@ module Combat exposing (Model, Msg, init, update, view)
 import Creature exposing (Attack, Creature, CreatureType(..))
 import Html exposing (Html, button, div, text)
 import Html.Events exposing (onClick)
+import List.Extra as ListX
 
 
 -- Model
 
 
 type alias Model =
-    { turnHistory : List ( Int, Attack )
+    { turnHistory : List Attack
     , players : List String
     , characters :
         { allies : List Creature
         , enemies : List Creature
         }
     , turn : Int
-    , turnOrder : List ( Int, Creature )
+    , turnOrder : List Creature
```

Comment:
The Ints in the lists in the model were redundant, and it was easier to do updates without them, so I opted to take them out.  They can always be derived again by using `indexedMap`.

In general in Elm, redundant data in the model is usually a warning sign. Because everything is immutable, repeated data can get out of sync, in different places, which leads to bugs when different pieces of code work from different sources. It's usually better to have a more normalized model, and derive secondary data where you need it.

There's a bit more reduction in data we could do here, but I didn't want to throw in too many changes at once and obscure what I did.